src/arrayMethodIndexOf.js

```diff
@@ -5,8 +5,17 @@
  *
  */
 function applyCustomIndexOf() {
-  [].__proto__.indexOf2 = function(searchElement, fromIndex) {
-    // write code here
+  [].__proto__.indexOf2 = function(searchElement, fromIndex = 0) {
+    const arrayLength = this.length;
+    if (arrayLength === 0 ||
+      fromIndex >= this.length ||
+      searchElement === undefined) {
+      return -1;
+    }
+    for (let i = fromIndex; i < this.length; i++) {
```

Comment:
use variable `arrayLength ` if you already have it

Reply:
done